definitions/npm/styled-components_v4.x.x/flow_v0.92.0-/styled-components_v4.x.x.js

```diff
@@ -0,0 +1,443 @@
+    pre: React$ElementRef<'pre'>,
+    progress: React$ElementRef<'progress'>,
+    q: React$ElementRef<'q'>,
+    rp: React$ElementRef<'rp'>,
+    rt: React$ElementRef<'rt'>,
+    ruby: React$ElementRef<'ruby'>,
+    s: React$ElementRef<'s'>,
+    samp: React$ElementRef<'samp'>,
+    script: React$ElementRef<'script'>,
+    section: React$ElementRef<'section'>,
+    select: React$ElementRef<'select'>,
+    small: React$ElementRef<'small'>,
+    source: React$ElementRef<'source'>,
+    span: React$ElementRef<'span'>,
+    strong: React$ElementRef<'strong'>,
+    style: React$ElementRef<'style'>,
+    sub: React$ElementRef<'sub'>,
+    summary: React$ElementRef<'summary'>,
+    sup: React$ElementRef<'sup'>,
+    table: React$ElementRef<'table'>,
+    tbody: React$ElementRef<'tbody'>,
+    td: React$ElementRef<'td'>,
+    textarea: React$ElementRef<'textarea'>,
+    tfoot: React$ElementRef<'tfoot'>,
+    th: React$ElementRef<'th'>,
+    thead: React$ElementRef<'thead'>,
+    time: React$ElementRef<'time'>,
+    title: React$ElementRef<'title'>,
+    tr: React$ElementRef<'tr'>,
+    track: React$ElementRef<'track'>,
+    u: React$ElementRef<'u'>,
+    ul: React$ElementRef<'ul'>,
+    var: React$ElementRef<'var'>,
+    video: React$ElementRef<'video'>,
+    wbr: React$ElementRef<'wbr'>,
+
+    // SVG
+    circle: React$ElementRef<'circle'>,
+    clipPath: React$ElementRef<'clipPath'>,
+    defs: React$ElementRef<'defs'>,
+    ellipse: React$ElementRef<'ellipse'>,
+    g: React$ElementRef<'g'>,
+    image: React$ElementRef<'image'>,
+    line: React$ElementRef<'line'>,
+    linearGradient: React$ElementRef<'linearGradient'>,
+    mask: React$ElementRef<'mask'>,
+    path: React$ElementRef<'path'>,
+    pattern: React$ElementRef<'pattern'>,
+    polygon: React$ElementRef<'polygon'>,
+    polyline: React$ElementRef<'polyline'>,
+    radialGradient: React$ElementRef<'radialGradient'>,
+    rect: React$ElementRef<'rect'>,
+    stop: React$ElementRef<'stop'>,
+    svg: React$ElementRef<'svg'>,
+    text: React$ElementRef<'text'>,
+    tspan: React$ElementRef<'tspan'>,
+
+    // Deprecated, should be HTMLUnknownElement, but Flow doesn't support it
+    keygen: React$ElementRef<'keygen'>,
+    menuitem: React$ElementRef<'menuitem'>,
+  }
+
+  declare type BuiltinElementType<ElementName: string> = $ElementType<BuiltinElementInstances, ElementName>
+
+  declare class InterpolatableComponent<P> extends React$Component<P> {
+    static +styledComponentId: string;
+  }
+
+  declare export type Interpolation<P> =
+                                       | ((executionContext: P) => Interpolation<any>) // eslint-disable-line flowtype/no-weak-types
+                                       | InterpolatableComponent<any> // eslint-disable-line flowtype/no-weak-types
+                                       | CSSRules
+                                       | KeyFrames
+                                       | string
+                                       | number
```

Comment:
I think types like `boolean` is missing here. Consider the following styled component:

```
type EnabledElementPropsType = {|selected: boolean, children: React.Node, theme: ThemeType|}

export const EnabledElement: StyledComponent<EnabledElementPropsType, ThemeType, *> = styled(Element)`
  color: ${props => props.theme.textColor};
  ${props => props.selected && `background-color: ${props.theme.colors.backgroundAccentColor}`};
`
```

I think it should be allowed to add a prop with the type "boolean".

Reply:
These are not _props_ interpolations that styled components understands (when using the tagged template literal syntax). So, this doesn't make sense to styled components (AFAIK):
~~~js
const Something = styled.div`
  ${true}
`
~~~

Reply:
You are right, I had to change my code to `  ${props => props.selected ? `background-color: ${props.theme.colors.backgroundAccentColor}` : ''};` which makes a lot more sense.Thank you!